### PR TITLE
Add reverse complement process

### PR DESCRIPTION
**Pull request by AnzeLovse**
## Checklist

* [ ] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [ ] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [ ] All inputs are used in process.
* [ ] All output fields have their ``re-save`` calls.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of tool should be self explanatory with links to tool's homepage
  and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required: false`` is not needed.
* Add descriptive error and warning messages.
* Place a ``re-checkrc`` call after each process step (call of a tool, a
  Bash command, a Python script or an R script).
* Add ``re-progress`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Changes to Docker images need to be accepted in a separate pull request.
  Before the new images can be used for automated testing of your PR, they
  must be manually tagged and pushed to Docker Hub.
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files with spaces:
  * Genomes: ``genome species.fa.gz``
  * Annotations: ``annotation species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Output and specific input files should include the name of the tested tool.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if length of some list is 0, if string is given where one expects
  integer, if file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 16384,
+        },
+    }
+    data_name = '{{ reads|sample_name|default("?") }}'
+    version = '1.0.0'
+
+
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Too many blank lines.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+"""Reverse complement reads with Seqtk."""
+import os
+
+from resolwe.process import Process, DataField, FileField, Cmd, ListField, StringField
+
+
+class ReverseComplementSingle(Process):
+    """Reverse complement FASTQ reads file using Seqtk."""
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Comment could perhaps include that this process is for single-end.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    slug = 'seqtk-rev-complement-single'
+    process_type = 'data:reads:fastq:single:seqtk:comp'
+    name = 'Reverse complement FASTQ'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Name could include that this is a single-end process. My suggestion would be to use `(single-end)`, as do other processes in resolwe-bio.
Screenshot from 2019-11-04 21-05-38

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq]] > 'input_reads.fastq.gz')()
+
+        # Reverse complement reads
+        (Cmd['seqtk']['seq', '-r', 'input_reads.fastq.gz'] > f'{name}_complemented.fastq')()
+
+        (Cmd['gzip'][f'{name}_complemented.fastq'])()
+
+        # Save the outputs
+        outputs.fastq = [f'{name}_complemented.fastq.gz']
+
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Add one more blank line.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+class ReverseComplementPaired(Process):
+    """Reverse complement FASTQ reads file using Seqtk."""
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Comment could include that this process works on paired-end data.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+    slug = 'seqtk-rev-complement-paired'
+    process_type = 'data:reads:fastq:paired:seqtk:comp'
+    name = 'Reverse complement FASTQ (Paired)'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: My suggestion would be to use `(paired-end)` as do other processes on the platform.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image' : 'resolwebio/common:1.1.0'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Redundant space in front of the colon.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+    class Input:
+        """Input fields to process ReverseComplementPaired."""
+
+        reads = DataField('reads:fastq:paired', label='Reads')
+        select_mate = StringField(
+            label='Select mate',
+            description='Select the which mate should be reverse complemented.',
+            choices=[
+                ('Mate 1', 'Mate 1'),
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Would computer readable name be in any way better, i.e. `('mate1', 'Mate 1')`?

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+                ('Mate 2', 'Mate 2'),
+                ('Both', 'Both')
+            ],
+            default='Mate 1',
+        )
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label='Reverse complemented FASTQ file')
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: I would name it `fastq1` just to be a little bit more explicit as you seem to follow the 1/2 convention in other variable names.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        elif inputs.select_mate == 'Mate 2':
+            # Reverse complement mate 2
+            (Cmd['seqtk']['seq', '-r', f'{name_mate2}_original.fastq.gz'] > f'{name_mate2}_complemented.fastq')()
+            (Cmd['gzip'][f'{name_mate2}_complemented.fastq'])()
+
+            # Save the outputs
+            outputs.fastq = [f'{name_mate1}_original.fastq.gz']
+            outputs.fastq2 = [f'{name_mate2}_complemented.fastq.gz']
+
+        else:
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Personally I would explicitly do three `if` statements, something along the lines of

```
if inputs.select_mate == "Mate 1":
    pass

if inputs.select_mate == "Mate 2":
    pass

if inputs.select_mate == "Both":
    pass
```
... just because it explicitly speaks to choices in the process preamble. I had to take a few minutes and see what the last `else` does (because there's a lot of repeating code, differing in small way between `if/else` clauses) because I wasn't very diligent in reading what choices there are. Once I understood the code I checked the `select_mate` options, it made sense.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+            # Reverse complement mate 1
+            (Cmd['seqtk']['seq', '-r', f'{name_mate1}_original.fastq.gz'] > f'{name_mate1}_complemented.fastq')()
+            (Cmd['gzip'][f'{name_mate1}_complemented.fastq'])()
+            
+            # Reverse complement mate 2
+            (Cmd['seqtk']['seq', '-r', f'{name_mate2}_original.fastq.gz'] > f'{name_mate2}_complemented.fastq')()
+            (Cmd['gzip'][f'{name_mate2}_complemented.fastq'])()
+
+            # Save the outputs
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Comment IMO redundant.

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 12]
```diff
@@ -548,3 +548,33 @@ def test_umi_tools_dedup(self):
...
+            }
+
+            revcomp_single = self.run_process('seqtk-rev-complement-single', inputs_single)
+
+            self.assertFiles(revcomp_single, 'fastq', ['hs_slamseq_R1_complemented.fastq.gz'],
+                             compression='gzip')
+
+            inputs_paired = {
+                'reads' : paired_reads.id,
+                'select_mate' : 'Mate 1'
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_support_processors.py]: Redundant space in front of the colon.

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 13]
```diff
@@ -548,3 +548,33 @@ def test_umi_tools_dedup(self):
 
         dedup_paired = self.run_process('umi-tools-dedup', {'alignment': bam_paired.id})
         self.assertFile(dedup_paired, 'stats', './corall/output/dedup_paired_stats.txt')
+
+    @tag_process('seqtk-rev-complement-single', 'seqtk-rev-complement-paired')
+    def test_reverse_complement(self):
+        with self.preparation_stage():
+            reads = self.run_process('upload-fastq-single', {
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_support_processors.py]: Use `single_reads` to discern it explicitly from `paired_reads`.

Comment (romunov) [inline resolwe_bio/tests/processes/test_support_processors.py]: You can use `self.prepare_reads` helper function which will match your usage of `self.prepare_paired_reads`.

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 14]
```diff
@@ -548,3 +548,33 @@ def test_umi_tools_dedup(self):
...
+                'reads': reads.id
+            }
+
+            revcomp_single = self.run_process('seqtk-rev-complement-single', inputs_single)
+
+            self.assertFiles(revcomp_single, 'fastq', ['hs_slamseq_R1_complemented.fastq.gz'],
+                             compression='gzip')
+
+            inputs_paired = {
+                'reads' : paired_reads.id,
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_support_processors.py]: Redundant space in front of the colon.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+    """Reverse complement FASTQ reads file using Seqtk."""
+
+    slug = 'seqtk-rev-complement-single'
+    process_type = 'data:reads:fastq:single:seqtk:comp'
+    name = 'Reverse complement FASTQ'
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image' : 'resolwebio/common:1.1.0'
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Redundant space in front of the colon.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+import os
+
+from resolwe.process import Process, DataField, FileField, Cmd, ListField, StringField
+
+
+class ReverseComplementSingle(Process):
+    """Reverse complement FASTQ reads file using Seqtk."""
+
+    slug = 'seqtk-rev-complement-single'
+    process_type = 'data:reads:fastq:single:seqtk:comp'
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: I think `data:reads:fastq:single:seqtk` should suffice. Other read processing processes have types:

- `data:reads:fastq:single:bbduk`
- `data:reads:fastq:single:cutadapt`
- ...

The point of process type is to restrict data usage for upstream processes. IMO, one level od depth is enough, but i might be missing something.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 15 above)
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Any reason not to use the [latest version](https://hub.docker.com/r/resolwebio/common/tags) of `resolwebio/common`?

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image' : 'resolwebio/common:1.1.0'
+            },
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 16384,
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: What is the memory footprint of this process? We will be dealing with really large files ~10GB and more, so make sure that reserved memory is sufficient. It is difficult to perform such test locally, but did you maybe checked how memory requirements grow when increasing file size? Linearly, constant?

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+    class Input:
+        """Input fields to process ReverseComplementSingle."""
+
+        reads = DataField('reads:fastq:single', label='Reads')
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label='Reverse complemented FASTQ file')
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: We might want to add same outputs as in [other processes](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/import_data/seq_reads.yml#L33) of type `data:reads:fastq`, i.e `fastqc_url`, `fastqc_archive`. This is the whole point of processes that share the same process type: to have the same set of outputs.
@jkokosar an comment more on this.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        name = os.path.basename(inputs.reads.fastq[0].path).strip('.fastq.gz')
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Very dangerous. Check what is the output of `'gaaaaaz_my_fastq.fastq.gz'.strip('fastq.gz')`.
One way of handling this::

    basename = os.path.basename(inputs.reads.fastq[0].path)
    assert basename.endswith('fastq.gz')
    name = basename[:-8]

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+
+        # Concatenate multilane reads
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Usually, comments are not describing what code does, but why it does if not obvious from the context. If it is obvious what the code does (it should be, 99% of the time if code is well written) the comment is unnecessary. IMO in this and the cases below it is safe to remove the comments.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 8 above)
Comment (JureZmrzlikar) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Not really. We use the same notation in [other places](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/descriptors/sample.yml#L16) too.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 9 above)
Comment (JureZmrzlikar) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: I disagree. This is consistent with other processes of the same type, for example  [upload-fastq-paired](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/import_data/seq_reads.yml#L136)

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label='Reverse complemented FASTQ file')
+        fastq2 = ListField(FileField(), label='Remaining mate')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        name_mate1 = os.path.basename(inputs.reads.fastq[0].path).strip('.fastq.gz')
```
Comment (JureZmrzlikar) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: See comment above.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 10 above)
Comment (JureZmrzlikar) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: I disagree. You suggestion implies that all three options could be true at the same time. The nature of if-elif-else is handling three exclusive disjunct options and this is exactly what we are handling with.

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 22]
```diff
@@ -548,3 +548,33 @@ def test_umi_tools_dedup(self):
...
+    def test_reverse_complement(self):
+        with self.preparation_stage():
+            reads = self.run_process('upload-fastq-single', {
+                'src': ['hs_slamseq_R1.fastq.gz']
+            })
+
+            paired_reads = self.prepare_paired_reads(['hs_slamseq_R1.fastq.gz'],
+                                                     ['hs_slamseq_R2.fastq.gz'])
+
+            inputs_single = {
```
Comment (JureZmrzlikar) [inline resolwe_bio/tests/processes/test_support_processors.py]: It makes sense to put inputs close to the process that uses them. I suggest the following syntax:

    revcomp_single = self.run_process('seqtk-rev-complement-single', {'reads': reads.id})

and:

    revcomp_paired = self.run_process('seqtk-rev-complement-paired', {
        'reads' : paired_reads.id,
        'select_mate' : 'Mate 1',
    })

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 23]
```diff
@@ -548,3 +548,33 @@ def test_umi_tools_dedup(self):
...
+            self.assertFiles(revcomp_single, 'fastq', ['hs_slamseq_R1_complemented.fastq.gz'],
+                             compression='gzip')
+
+            inputs_paired = {
+                'reads' : paired_reads.id,
+                'select_mate' : 'Mate 1'
+            }
+
+            revcomp_paired = self.run_process('seqtk-rev-complement-paired', inputs_paired)
+            self.assertFiles(revcomp_paired, 'fastq', ['hs_slamseq_R1_complemented.fastq.gz'],
```
Comment (JureZmrzlikar) [inline resolwe_bio/tests/processes/test_support_processors.py]: This line can fit into 119 chars limit, which is the limit for this repo. It is more compact and easier to read if it is just in one line IMO. But this is a matter of style, so i leave it up to you.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 10 above)
Comment (romunov) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: My reasoning was that this is handled on the `StringField` as I am assuming only one choice can be selected at a time.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py (hunk 17 above)
Comment (AnzeLovse) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: I was only able to test it locally on read files with 5GB size.  According to the results of [this](https://bioinf.shenwei.me/seqkit/benchmark/) benchmarking the RAM usage is usually smaller than the file size.

Comment (JureZmrzlikar) [reply resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: OK.

File: resolwe_bio/tests/processes/test_support_processors.py [hunk 24]
```diff
@@ -548,3 +548,23 @@ def test_umi_tools_dedup(self):
...
         self.assertFile(dedup_paired, 'stats', './corall/output/dedup_paired_stats.txt')
+
+    @tag_process('seqtk-rev-complement-single', 'seqtk-rev-complement-paired')
+    def test_reverse_complement(self):
+        with self.preparation_stage():
+            single_reads = self.prepare_reads(['hs_slamseq_R1.fastq.gz'])
+
+            paired_reads = self.prepare_paired_reads(['hs_slamseq_R1.fastq.gz'], ['hs_slamseq_R2.fastq.gz'])
+
+            revcomp_single = self.run_process('seqtk-rev-complement-single', {'reads': single_reads.id})
```
Comment (JureZmrzlikar) [inline resolwe_bio/tests/processes/test_support_processors.py]: This should be one indentation level less than it is. The `preparation_stage` context manager only prepares input data objects, but the process that is tested should be put outside of it (and also all assert statements that follow it).

File: docs/CHANGELOG.rst [hunk 25]
```diff
@@ -13,6 +13,7 @@ Unreleased
 Added
 -----
 - Add ``resolwebio/slamdunk`` Docker image
+- Add ``seqtk-rev-complement-single`` process
```
Comment (jkokosar) [inline docs/CHANGELOG.rst]: Two processes were added, so modify to something like: Add ``seqtk-rev-complement-single`` and ``seqtk-rev-complement-paired`` processes

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 26]
```diff
@@ -0,0 +1,185 @@
...
+        """Run the analysis."""
+        basename = os.path.basename(inputs.reads.fastq[0].path)
+        assert basename.endswith('.fastq.gz')
+        name = basename[:-9]
+
+        # Concatenate multilane reads
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq]] > 'input_reads.fastq.gz')()
+
+        # Reverse complement reads
+        (Cmd['seqtk']['seq', '-r', 'input_reads.fastq.gz'] > f'{name}_complemented.fastq')()
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Think about putting `f'{name}_complemented.fastq'` into a variable as it repeats several times.

File: resolwe_bio/processes/support_processors/seqtk_reverse_complement.py [hunk 27]
```diff
@@ -0,0 +1,185 @@
...
+        assert basename_mate1.endswith('.fastq.gz')
+        assert basename_mate2.endswith('.fastq.gz')
+        name_mate1 = basename_mate1[:-9]
+        name_mate2 = basename_mate2[:-9]
+
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq]] > f'{name_mate1}_original.fastq.gz')()
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq2]] > f'{name_mate2}_original.fastq.gz')()
+
+        if inputs.select_mate == 'Mate 1':
+            (Cmd['seqtk']['seq', '-r', f'{name_mate1}_original.fastq.gz'] > f'{name_mate1}_complemented.fastq')()
```
Comment (romunov) [inline resolwe_bio/processes/support_processors/seqtk_reverse_complement.py]: Think about adding output strings into variables and use those throughout the rest of the process.

**Comment by jkokosar** [comment]
Please rebase to the latest master branch

**Comment by jkokosar** [comment]
Please rebase to the latest master. The ROSE2 issue has been fixed.
